testbed/tests/app/test_desktop.py

```diff
@@ -201,19 +203,19 @@ async def test_presentation_mode(app, app_probe, main_window, main_window_probe)
         screen_window_dict[window_information["paired_screen"]] = window_information[
             "window"
         ]
-
-    # Add delay to ensure windows are visible after animation.
+    # Wait for window animation before assertion.
     await main_window_probe.wait_for_window("All Test Windows are visible")
 
     # Enter presentation mode with a screen-window dict via the app
     app.enter_presentation_mode(screen_window_dict)
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
-        "App is in presentation mode", full_screen=True
-    )
-    assert app.in_presentation_mode
+
```

Comment:
Why is the test changing? The test shouldn't be changing *at all* if all we're doing is changing how the test is evaluated.

Reply:
I have only changed wait_for_window() to use the new parameter. The reason for why I have moved the wait_for_window() call to inside the for loop is because, we need to wait for each window to transition to the expected state. The polling mechanism checks each window individually to confirm its state transition.

Reply:
Ok, but as a result of this change, you're *not* waiting for `main_window` any more. Adding the extra window checks makes sense, but *removing* the main window check seems like a fairly major change in behavior.

Reply:
Previously, wait_for_window() was pausing the test for a fixed delay, and so it didn't matter on which window probe wait_for_window() was called. Hence, I had used main_window probe for wait_for_window(). But, now since wait_for_window() checks each window individually to confirm its state transition, so waiting for main_window isn't required anymore.

Reply:
Yes, but main window *isn't* in `window_list` - so the main window isn't being checked any more. A check that was previously being performed *isn't* being performed now. If the main_window is already in the right state, it should be a no-op.

Reply:
I have reverted the `main_window_probe.wait_for_window()` call, but `assert app.in_presentation_mode` needs to be inside the loop, as this condition will be true when the specified window polling for the new window state is completed.

Reply:
I have reverted the call to `main_window_probe.wait_for_window()`, but the `assert app.in_presentation_mode` assertion needs to be inside the loop. This is because the assertion will be true only when the polling mechanism has completed waiting for the specified window to transition into the new state.